nemo_aligner/data/nlp/datasets.py

```diff
@@ -344,16 +348,63 @@ def encode(self, text, append_eod=False):
 
         return text_ids, len(text_ids)
 
+    def _convert_messages(self, input_list):  # TODO: (@adithyare) this method should live elsewhare..
+        output_dict = {
+            "system": "",
+            "conversations": [],
+            "mask": "User",
+            "type": "VALUE_TO_TEXT",
+        }
+
+        # Extract the system message
+        for msg in input_list:
+            if msg["role"] == "system":
+                output_dict["system"] = msg["content"]
+                break  # Assuming only one system message
+
+        # Build the conversations list
+        for msg in input_list:
+            if msg["role"] != "system":
+                conversation_entry = {
+                    "from": msg["role"].capitalize(),  # Capitalize 'user' and 'assistant'
+                    "value": msg["content"],
+                    "label": None,
+                }
+                output_dict["conversations"].append(conversation_entry)
+
```

Comment:
if the user gets here and the input didn't have the keys you expect, do you think it'd help to add some assertions here to make sure the code doesn't error somewhere else, when the issue was they didn't pass the right input?